web/modules/custom/dpl_redia_legacy/src/Controller/RssFeeds/EventsController.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace Drupal\dpl_redia_legacy\Controller\RssFeeds;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\File\FileUrlGeneratorInterface;
+use Drupal\Core\Url;
+use Drupal\file\FileInterface;
+use Drupal\image\ImageStyleInterface;
+use Drupal\media\MediaInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use function Safe\filesize;
+use function Safe\getimagesize;
+
+/**
+ * Building a Redia-legacy RSS feed, showing eventinstances.
+ */
+class EventsController extends ControllerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    protected FileUrlGeneratorInterface $fileUrlGenerator,
+    protected RequestStack $requestStack,
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('file_url_generator'),
+      $container->get('request_stack'),
+
+    );
+  }
+
+  /**
+   * Getting the RSS/XML feed of the items.
+   */
+  public function getFeed(): Response {
+    $items = $this->getItems();
+
+    $rss_content = $this->buildRss($items);
+
+    $response = new Response();
+    $response->setContent($rss_content);
+    $response->headers->set('Content-Type', 'application/rss+xml');
+    return $response;
```

Comment:
Please update this to return a cacheable response.

I do not know how often the feed is requested but I do not see a reason why we should not be able to cache it in Varnish.

Reply:
As I see it you have changes the implementation to return a cacheable response but please also add some cacheable dependencies to make it behave right :smile: